### PR TITLE
Central bunch spacing producer

### PR DESCRIPTION
**Pull request by davidlange6**
attempt two at making a PR for a new module to centralize the bunch spacing in reco/analysis. This time I've included a parameter to allow prompt to override the bunch spacing in configuration based on the processing scenario being used.

Instead of storing this output, I suggest that this module just be run as needed. I've added it to the reco sequence, which should be sufficient for all current consumers. Eventually we'll need it in PAT as well.

### CODE REVIEW

**Comment by lgray** [comment]
@davidlange6 is it already needed in PAT (for regressions on the fly)

**Comment by lgray** [comment]
@davidlange6 Ah, see what you mean. There are no hardcoded instances of it. I'll send you a PR with the appropriate python updates.

**Comment by davidlange6** [comment]
please test

**Comment by davidlange6** [comment]
ah - right, all of the other reco sequences… will update.

**Comment by davidlange6** [comment]
@cmsbuild, please test

File: Configuration/StandardSequences/python/PAT_cff.py [hunk 1]
```diff
@@ -3,5 +3,6 @@
 from PhysicsTools.PatAlgos.producersLayer1.patCandidates_cff import *
 from PhysicsTools.PatAlgos.selectionLayer1.selectedPatCandidates_cff import *
 from PhysicsTools.PatAlgos.slimming.slimming_cff import *
+from RecoLuminosity.LumiProducer.bunchSpacingProducer_cfi import *
```
Comment (lgray) [inline Configuration/StandardSequences/python/PAT_cff.py]: Yep, that'll do just fine too. Thanks.

**Comment by davidlange6** [comment]
cmbuild, please test

**Comment by davidlange6** [comment]
please test

**Comment by slava77** [comment]
@degano @smuzaffar could you please check the error message reporting: either the style has changed 
or there is a problem with the cms-bot setup
https://github.com/cms-sw/cmssw/pull/11544#issuecomment-144458920

**Comment by smuzaffar** [comment]
pleas test

looks like my last changes to set timeout for unit tests did not go well

**Comment by smuzaffar** [comment]
pleas test

looks like my last changes to set timeout for unit tests did not go well

**Comment by davidlange6** [comment]
identity in all of the tests. Good.

File: RecoLuminosity/LumiProducer/plugins/BunchSpacingProducer.cc [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      bunchSpacing = 25;
+    }
+  }
+  else{
+    edm::Handle<int> bunchSpacingH;
+    e.getByToken(bunchSpacing_,bunchSpacingH);
+    bunchSpacing = *bunchSpacingH;
+  }
+
+  std::cout << "Derived from run number " << bunchSpacing << std::endl;
```
Comment (slava77) [inline RecoLuminosity/LumiProducer/plugins/BunchSpacingProducer.cc]: cout, really?

Comment (davidlange6) [inline RecoLuminosity/LumiProducer/plugins/BunchSpacingProducer.cc]: debug printout… whoops

**Pull request merged**